src/Raven.Server/Documents/PeriodicBackup/EncryptedBackup.cs

```diff
@@ -190,34 +191,36 @@ public unsafe EncryptingXChaCha20Poly1305Stream(Stream inner, byte[] key)
 
         public override void Flush()
         {
-            if (_pos > 0)
+            // no-op - because we *cannot* do anything here, we *require* that all writes, except the last one
+            // will be on 4Kb aligned value. This is already handled, but we may want to flush past writes, so
+            // will allow it
+            if (_shouldFlush)
             {
-                var sizeToWrite = EncryptBuffer();
-                _inner.Write(_cyrptedBuffer, 0, sizeToWrite);
-                _pos = 0;
+                _inner.Flush();
+                _shouldFlush = false;
             }
-            _inner.Flush();
         }
 
         public override async Task FlushAsync(CancellationToken cancellationToken)
         {
-            if (_pos > 0)
+            // no-op - because we *cannot* do anything here, we *require* that all writes, except the last one
+            // will be on 4Kb aligned value. This is already handled, but we may want to flush past writes, so
+            // will allow it
+            if (_shouldFlush)
             {
-                var sizeToWrite = EncryptBuffer();
-                await _inner.WriteAsync(_cyrptedBuffer, 0, sizeToWrite, cancellationToken);
-                _pos = 0;
+                await _inner.FlushAsync(cancellationToken);
+                _shouldFlush = false;
             }
-            await _inner.FlushAsync(cancellationToken);
         }
 
         public void Flush(bool flushToDisk)
         {
-            if (_pos > 0)
-            {
-                var sizeToWrite = EncryptBuffer();
-                _inner.Write(_cyrptedBuffer, 0, sizeToWrite);
-                _pos = 0;
-            }
+            if (_shouldFlush == false)
+                return;
+            _shouldFlush = false;
```

Comment:
shouldn't we first attempt to flush then set this to false?

Reply:
I don't think it matters, if this throws (only way that matter) and then the entire thing is broken anyway